packages/stryker-html-reporter/src/HtmlReporter.ts

```diff
@@ -1,26 +1,28 @@
-import { getLogger } from 'stryker-api/logging';
+import { Logger } from 'stryker-api/logging';
 import fileUrl = require('file-url');
 import * as path from 'path';
-import { Config } from 'stryker-api/config';
 import { Reporter, MutantResult, SourceFile, ScoreResult } from 'stryker-api/report';
 import * as util from './util';
 import * as templates from './templates';
 import Breadcrumb from './Breadcrumb';
+import { StrykerOptions } from 'stryker-api/core';
+import { tokens, commonTokens } from 'stryker-api/plugin';
 
 const DEFAULT_BASE_FOLDER = path.normalize('reports/mutation/html');
 export const RESOURCES_DIR_NAME = 'strykerResources';
 
 export default class HtmlReporter implements Reporter {
-  private readonly log = getLogger(HtmlReporter.name);
-  private _baseDir: string;
-  private mainPromise: Promise<void>;
-  private mutantResults: MutantResult[];
-  private files: SourceFile[];
-  private scoreResult: ScoreResult;
-
-  constructor(private readonly options: Config) {
+  private _baseDir!: string;
```

Comment:
What does this `!` do?

Reply:
It's the [definite assignment operator](https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#definite-assignment-assertions). This allows us to use `--strictPropertyInitialization`, while not assigning them in the constructor.